compiler-core/src/error.rs

```diff
@@ -180,18 +180,33 @@ pub enum Error {
     #[error("{0}")]
     Http(String),
 
+    #[error("Git dependencies are currently unsupported")]
+    GitDependencyUnsuported,
+
+    #[error("Failed to create canonical path for package {0}")]
+    DependencyCanonicalizationFailed(String),
+
     #[error("Dependency tree resolution failed: {0}")]
     DependencyResolutionFailed(String),
 
     #[error("The package {0} is listed in dependencies and dev-dependencies")]
     DuplicateDependency(String),
 
+    #[error("The package {0} is provided multiple times, as {1} and {2}")]
+    ProvidedDependencyConflict(String, String, String),
```

Comment:
Could we name these fields please? I can't tell what the second and third strings are.